packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js

```diff
@@ -110,30 +167,48 @@ export default /* @ngInject */ function IpFirewallCtrl(
                 options.push(result.tcpOption);
               }
 
-              $scope.rules.list.results[i].options = options.join('<br/>');
+              if (result.action === 'deny') {
+                isDenyRule = true;
+              }
+
+              self.rules.list.results[i].options = options.join('<br/>');
 
               // Go poll
               if (
-                result.state === 'CREATION_PENDING' ||
-                result.state === 'REMOVAL_PENDING'
+                result.state === self.constants.CREATION_PENDING ||
+                result.state === 'removalPending'
```

Comment:
**nitpick:** constant can be added for `removalPending` as well similar to `CREATION_PENDING`